array-destructuring/exercise-3/exercise.js

```diff
@@ -8,3 +8,14 @@ let order = [
 ];
 
 
+console.log("QTY      ITEM                 TOTAL");
+let totalPrice = 0;
+order.forEach(({quantity, itemName, unitPrice}) => {
+  const itemPrice = `${unitPrice * quantity}`;
+  console.log(`${quantity}\t${itemName}\t\t\t\t${itemPrice}`);
+  totalPrice = totalPrice + unitPrice * quantity;
+})
+console.log(`Total: ${totalPrice}`);
+
+
+// Question: Why doesn't it work when I  do totalPrice = totalPrice + itemPrice in line 16
```

Comment:
To answer your question about why doing `totalPrice = totalPrice + itemPrice;` does not work, it is because on line 14 you are defining `itemPrice` as a string by using [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals).

If you were to remove the template literals and just do `const itemPrice = unitPrice * quantity;`, then it would work.

Reply:
Also, an improvement you could make to this solution would be to use `Array.reduce()` to create a running aggregate instead of defining `totalPrice` outside of the loop.

Here is a link that explains using reduce to calculate a sum.

https://linuxhint.com/call-reduce-on-an-array-of-objects-to-sum-their-properties/

Reply:
Thank you very much for noticing my question and answering it! Thank you for the helpful article! I have made some changes.